src/variants/api/ymodem.rs

```diff
@@ -22,6 +38,20 @@ pub struct YModem {
     initial_errors: u32,
 }
 
+impl YModem {
+    fn add_error(&mut self) -> ModemResult<()> {
+        self.errors += 1;
+
+        if self.errors >= self.max_errors {
+            #[cfg(defmt)]
+            error!("Exhausted max retries ({}) while sending start frame in YMODEM transfer", self.max_errors);
+            return Err(ModemError::ExhaustedRetries { errors: self.max_errors });
+        } else {
+            Ok(())
+        }
```

Comment:
This branch can be simplified. Remove `else` and simply end function with `Ok(())`.